experimental/SymmetricIntersections/Representations.jl

```diff
@@ -0,0 +1,1465 @@
+  projectively faithful representations of a Schur cover associated to those
+  faithful projective representations
+"""
+function _has_pfr(G::T, dim::Int) where T <: Oscar.GAPGroup
+  # we start by computing a Schur cover and we turn it into an Oscar object
+  G_gap = G.X
+  f_gap = GG.EpimorphismSchurCover(G_gap)
+  H_gap = GG.Source(f_gap)
+  n, p = ispower(GG.Size(H_gap))
+  if isprime(p)
+    fff_gap = GG.EpimorphismPGroup(H_gap, p)
+    E_gap = fff_gap(H_gap)
+  else
+    fff_gap = GG.IsomorphismPermGroup(H_gap)
+    E_gap = fff_gap(H_gap)
+  end
+  E = Oscar._get_type(E_gap)(E_gap)
+  H = Oscar._get_type(H_gap)(H_gap)
+  fff = inv(GAPGroupHomomorphism(H, E, fff_gap))
+  f = GAPGroupHomomorphism(H, G, f_gap)
+  pschur = compose(fff, f)
+  @assert is_surjective(f)
+
+  # now we need to enumerate of characters of E of degree dim, modulo
+  # multiplication by a linear character and up to the bounds
+  # fixed by `setup`
+  RR = representation_ring(E)
+  ct = character_table_underlying_group(RR)
+  _Irr = irreducible_characters_underlying_group(RR)
+  _L = Int[Int(degree(irr)) for irr in _Irr]
+  ps = sortperm(_L)
+  L, Irr = _L[ps], _Irr[ps]
+  RR.irr = Irr
+  L = Int[i for i in L if i <= dim]
+  lins = Oscar.GAPGroupClassFunction[irr for irr in Irr if irr[1] == 1]
+  keep_index = Int[]
+  sum_index = Vector{Int}[]
+  chars = Oscar.GAPGroupClassFunction[]
+  local El::ElevCtx
+  El = elevator(L, dim)
+  for l in El
+    chi = sum(Irr[l])
+    if !is_projective(chi, pschur) || !is_faithful(chi, pschur)
+      continue
+    end
+    if any(lin -> lin*chi in chars, lins) # afforded by representations with similar reductions
+      continue
+    end
+    push!(chars, chi)
+    keep_index = union(keep_index, l)
+    push!(sum_index, l)
+  end
+  bool = length(keep_index) > 0
+  return bool, RR, sum_index, pschur
+end
+
+@doc Markdown.doc"""
+    faithful_projective_representations(o::Int, n::Int, dim::Int)
+    faithful_projective_representations(G::Oscar.GAPGroup, dim::Int)
+                                                             -> Vector{ProjRep}
+
+Given a small group `G` of ID `[o,n]`, return a set of representatives
+of equivalences classes of faithful projective representations of `G` of complex
+dimension `dim`.
+"""
+function faithful_projective_representations(G::Oscar.GAPGroup, dim::Int)
+  bool, RR, sum_index, p = _has_pfr(G, dim)
+  F = base_field(RR)
+  E = underlying_group(RR)
+  pfr = ProjRep{typeof(F), typeof(E), elem_type(F), typeof(p)}[]
+  if !bool
+    return pfr
+  end
+  Irr = irreducible_characters_underlying_group(RR)
+  for l in sum_index
+    chara = sum(Irr[l])
+    @assert chara[1] == dim
+    lr = affording_representation(RR, chara)
+    pr = ProjRep{typeof(F), typeof(E), elem_type(F), typeof(p)}(lr, p)
+    push!(pfr, pr)
+  end
+  return pfr
+end
+
+faithful_projective_representations(o::Int, n::Int, d::Int) = faithful_projective_representations(small_group(o, n), d)
```

Comment:
This seems overly specific. I would suggest to remove it (but of course then the tests and docstring needs to be adjusted)
```suggestion
```